core/src/main/scala/org/elasticmq/actor/QueueManagerActor.scala

```diff
@@ -7,18 +7,23 @@ import org.elasticmq.actor.reply._
 import org.elasticmq.msg._
 import org.elasticmq.util.{Logging, NowProvider}
 
+import scala.collection.mutable
 import scala.reflect._
 
 class QueueManagerActor(nowProvider: NowProvider, limits: Limits, queueEventListener: Option[ActorRef])
     extends ReplyingActor
+    with QueueManagerActorStorage
+    with QueueManagerMessageMoveOps
     with Logging {
+
   type M[X] = QueueManagerMsg[X]
   val ev: ClassTag[QueueManagerMsg[Unit]] = classTag[M[Unit]]
 
-  case class ActorWithQueueData(actorRef: ActorRef, queueData: QueueData)
-  private val queues = collection.mutable.HashMap[String, ActorWithQueueData]()
+  val queues: mutable.Map[MessageMoveTaskHandle, ActorWithQueueData] = mutable.HashMap[String, ActorWithQueueData]()
 
-  def receiveAndReply[T](msg: QueueManagerMsg[T]): ReplyAction[T] =
+  // TODO: create *Ops class like in QueueActor
```

Comment:
**question:** Is this supposed to be implemented in a subsequent PR with refactoring, to avoid extending the scope too much?

Reply:
Yes, this is for future me